charts/vector/docs/datadog_quickstart.yaml

```diff
@@ -0,0 +1,198 @@
+## See Vector helm documentation to learn more:
+## https://vector.dev/docs/setup/installation/package-managers/helm/
+
+## Create a Secret resource for Vector to use
+secrets:
+  # secrets.generic -- Each Key/Value will be added to the Secret's data key, each value should be raw and NOT base64 encoded
+  ## Any secrets can be provided here, it's commonly used for credentials and other access related values.
+  ## NOTE: Don't commit unencrypted secrets to git!
+  generic:
+    datadog_api_key: "REPLACE_ME"
+
+## Configure a HorizontalPodAutoscaler for Vector
+autoscaling:
+  enabled: true
+  minReplicas: 2
+  ## The provided HAProxy config is limited to 10 backends 
+  maxReplicas: 10
+  targetCPUUtilizationPercentage: 80
+
+podDisruptionBudget:
+  enabled: true
+  minAvailable: 1
+
+# env -- Set environment variables in Vector containers
+## The examples below leverage examples from secrets.generic and assume no name overrides with a Release name of "vector"
+env:
+  - name: DATADOG_API_KEY
+    valueFrom:
+      secretKeyRef:
+        name: vector
+        key: datadog_api_key
+
+# resources -- Set Vector resource requests and limits.
+resources:
+  ## Required for HPA to function
+  requests:
+    cpu: 1000m
+    memory: 512Mi
+  # limits:
+  #   cpu: 200m
+  #   memory: 256Mi
+
+# affinity -- Allow Vector to schedule using affinity rules
+## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+affinity:
+  ## Scale across different AZs by default.
+  podAntiAffinity:
+    preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+              - key: app.kubernetes.io/name
+                operator: In
+                values:
+                  - vector
+              - key: app.kubernetes.io/component
+                operator: In
+                values:
+                  - Aggregator
+          topologyKey: topology.kubernetes.io/zone
+
+# customConfig -- Override Vector's default configs, if used **all** options need to be specified
+## This section supports using helm templates to populate dynamic values
+## Ref: https://vector.dev/docs/reference/configuration/
+customConfig:
+  data_dir: /vector-data-dir
+  api:
+    enabled: true
+    address: 0.0.0.0:8686
+    playground: false
+  sources:
+    datadog_agent:
+      address: 0.0.0.0:8282
+      type: datadog_agent
+      multiple_outputs: true
+    internal_logs:
+      type: internal_logs
+    internal_metrics:
+      type: internal_metrics
+  transforms:
+    remap_logs:
+      type: remap
+      inputs:
+        - datadog_agent.logs
+        - internal_logs
+      source: |
+        # TODO: Having Datadog Agents collect Aggregator logs would provide better enrichment,
+        # internal_logs will also provide the Pod's name as the host facet in Datadog rather
+        # than the name of the Node
```

Comment:
It's probably a better experience on the Datadog side to remove the `console` sink and rely on `vector tap`.